packages/data-service/src/data-service.ts

```diff
@@ -2285,6 +2314,48 @@ class DataServiceImpl extends WithLogContext implements DataService {
     return { name, ...normalized };
   }
 
+  @op(mongoLogId(1_001_000_063))
+  async previewUpdate(
+    ns: string,
+    filter: Document,
+    update: Document
+  ): Promise<UpdatePreview> {
+    const coll = this._collection(ns, 'CRUD');
+    const session = this._startSession('CRUD');
+    try {
+      session.startTransaction();
+
+      const docsToPreview = await coll
+        .find(filter, { session })
+        .sort({ _id: 1 })
+        .limit(PREVIEW_SAMPLE)
+        .maxTimeMS(PREVIEW_TIMEOUT)
+        .toArray();
+
+      const idsToPreview = docsToPreview.map((doc) => doc._id);
+      await coll.updateMany({ _id: { $in: idsToPreview } }, update, {
+        session,
+        maxTimeMS: PREVIEW_TIMEOUT,
+      });
+      const changedDocs = await coll
+        .find({ _id: { $in: idsToPreview } }, { session })
+        .sort({ _id: 1 })
+        .maxTimeMS(PREVIEW_TIMEOUT)
```

Comment:
Do we want the timeout to be applied to the individual operations, or to the whole transaction? Intuitively, without reading the code here, I would probably have expected the latter